code/modules/unit_tests/changeling.dm

```diff
@@ -0,0 +1,95 @@
+/// Tests transformation sting goes back and forth correctly
+/datum/unit_test/transformation_sting
+	var/ling_name = "Is-A-Changeling"
+	var/base_victim_name
+	var/last_frame = 1
+	var/icon/final_icon
```

Comment:
user above me doesn't know about the icon cache

Reply:
what if you told me what it is instead of telling me i don't know what it is 👉 👈

Reply:
I thought jolly's pr (https://github.com/tgstation/tgstation/pull/78799) would get linked but it didn't so I'll just info dump here
Icons are basically wrappers around file references, they become modifiable while they are "cached", or loaded in.

The problem with this is byond was made by memory conscious devs, so an icon doesn't STAY cached forever.
There's a first in first out queue sitting in memory somewhere with a set length, and after X amount of icons are loaded, the oldest one gets dropped.

This'd ALSO be fine if you could treat dropped icons like normal ones, but you can't. Trying to modify them, like with the Insert() proc, will instead throw an error. Just says "bad icon" and goes about its day.

It's good to account for this when holding refs to icons, since it can cause some very confusing errors

The only way to avoid bad icon errors is to reload the icon, basically by doing something like this 
`final_icon = icon(final_icon)`

This reinserts it into the cache, and makes it safe to modify again.

Reply:
The real headache is this is documented fucking nowhere, not even in my writing, and hardly anyone (me and ninja I think) know about it.
Worst fucking runtime name in the history of the universe I swear man.